File: NotesAppJv/app/src/main/java/com/henriquesbraga/notesappjv/activity/main/adapter/MainAdapter.java
package com.henriquesbraga.notesappjv.activity.main.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.henriquesbraga.notesappjv.R;
import com.henriquesbraga.notesappjv.model.Note;

import java.util.List;

public class MainAdapter extends RecyclerView.Adapter<MainViewHolder> {

    private Context context;
    private List<Note> notes;
    private ItemClickListener itemClickListener;

    public MainAdapter(Context context, List<Note> notes, ItemClickListener itemClickListener) {
        this.context = context;
        this.notes = notes;
        this.itemClickListener = itemClickListener;
    }

    @NonNull
    @Override
    public MainViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_note, parent, false);
        return new MainViewHolder(view, itemClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull MainViewHolder holder, int position) {
        holder.bind(notes.get(position));
    }

    @Override
    public int getItemCount() {
        return notes.size();
    }
}

File: NotesAppKt/app/src/main/java/com/henriquesbraga/notesappkt/activity/main/adapter/MainAdapter.kt
package com.henriquesbraga.notesappkt.activity.main.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.henriquesbraga.notesappkt.R
import com.henriquesbraga.notesappkt.model.Note

class MainAdapter(context: Context, notes: ArrayList<Note>, itemClickListener: ItemClickListener): RecyclerView.Adapter<MainViewHolder>() {

    var context = context
    var notes = notes
    var itemClickListener = itemClickListener


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MainViewHolder {
        var view = LayoutInflater.from(context).inflate(R.layout.item_note, parent, false)
        return MainViewHolder(view, itemClickListener)
    }

    override fun onBindViewHolder(holder: MainViewHolder, position: Int) {
        holder.bind(notes[position])

    }

    override fun getItemCount(): Int {
        return notes.count()
    }

}